crates/astria-conductor/src/reader.rs

```diff
@@ -282,3 +349,235 @@ impl Reader {
         Ok(())
     }
 }
+
+fn verify_commit(commit: &Commit, validator_set: &ValidatorSet, chain_id: &str) -> Result<()> {
+    if commit.height != validator_set.block_height {
+        bail!(
+            "commit height mismatch: expected {}, got {}",
+            validator_set.block_height,
+            commit.height
+        );
+    }
+
+    let mut total_voting_power = 0u64;
+    validator_set
+        .validators
+        .iter()
+        .try_for_each(|v| -> Result<()> {
+            total_voting_power += v.voting_power.parse::<u64>()?;
```

Comment:
If voting power must be an integer, let's fix the type of `Validator` and make `voting_power: u64` so it already fails at ingestion, not deep into processing it.

Reply:
unfortunately, the RPC returns this as a string, eg `"1000"`, so serde fails to deserialize it into `u64`. I added a custom type + deserialize implementation for now

Reply:
ugh, that's terrible :(